app/models/ability.rb

```diff
@@ -0,0 +1,19 @@
+class Ability
+  include CanCan::Ability
+
+  def initialize(current_user)
+    @user = current_user || User.new
+
+    if @user.role == 'admin'
```

Comment:
- [ ] **[OPTIONAL]** It would be nice to convert this into a method (admin?) in the User model and then use the method here.

  For example in the User model you can define this method.
  ```
  def admin?
    role == 'admin'
  end
  ```

  Then in this `ability.rb` file you can use this:
  ```
  @user.admin?
  ```